sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractLanguageConfigurationTest.java

```diff
@@ -73,6 +73,36 @@ public void vulnerabilityCategories_reads_configuration() {
     assertThat(config.vulnerabilityCategories()).containsExactly("A", "B", "C");
   }
 
+  @Test
+  public void analyzeGeneratedCodeIsTrue() {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getBoolean("sonar.cs.analyzeGeneratedCode")).thenReturn(Optional.of(true));
+
+    AbstractLanguageConfiguration config = createConfiguration(configuration);
+
+    assertThat(config.analyzeGeneratedCode()).isTrue();
+  }
+
+  @Test
+  public void analyzeGeneratedCodeIsFalse() {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getBoolean("sonar.cs.analyzeGeneratedCode")).thenReturn(Optional.of(false));
+
+    AbstractLanguageConfiguration config = createConfiguration(configuration);
+
+    assertThat(config.analyzeGeneratedCode()).isFalse();
+  }
+
+  @Test
+  public void analyzeGeneratedCodeIsEmpty() {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getBoolean("sonar.cs.analyzeGeneratedCode")).thenReturn(Optional.empty());
```

Comment:
Can this line be removed for authenticity? In normal scenario, the line would be missing and reading of the configuration will deal with the `Optional` stuff.

I would not add another test.

Reply:
Well, it's a unit test, not an integration test. The part about reading the configuration is at integration level.

I will leave it as, especially as the code has inside the `orElse` part

```
    return configuration.getBoolean(AbstractPropertyDefinitions.getAnalyzeGeneratedCode(languageKey)).orElse(false);
```

If the configuration returns `empty`, the `orElse` is executed, and we need a test for that.